XpressAI/xircuits
# ✨Add rendering latest node option

# Description

This will add another option in `NodeActionPanel` to re-render node, getting its latest ports configuration. Also, refactor general and advanced component library.

## Pull Request Type


## Type of Change


# Tests

1. (Long method)
   1. Add any node into canvas
   2. Change ports from its script
   3. Do `ctrl+left-click` on the node 
   4. Choose `Render Node`
   5. It should render the node accordingly
2. (Short method)
   1. Open any old example
   2. Do `ctrl+left-click` on the node which you're sure there's some change
   3. Choose `Render Node`
   4. It should render the node accordingly

## Tested on?
